src/System.Text.Encoding/tests/NegativeEncodingTests.cs

```diff
@@ -256,5 +256,65 @@ public static void GetString_Invalid(Encoding encoding)
             Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 1, 1));
             Assert.Throws<ArgumentOutOfRangeException>("bytes", () => encoding.GetString(new byte[1], 0, 2));
         }
+
+        public static unsafe void Encode_Invalid(Encoding encoding, string chars, int index, int count)
+        {
+            char[] charsArray = chars.ToCharArray();
+            byte[] bytes = new byte[encoding.GetMaxByteCount(count)];
+
+            if (index == 0 && count == chars.Length)
+            {
+                Assert.Throws<EncoderFallbackException>(() => encoding.GetByteCount(chars));
```

Comment:
This test assumes the input encoding will have encoding fallback that throws exception. I know you are instantiating the UTF8 encoding with the flag does that but this also means any other encodings passed here (in the future) should do the same. I would suggest you assert encoding.EncoderFallback is equal to EncoderFallback.ExceptionFallback